pkg/pgmodel/ingestor/copier.go

```diff
@@ -332,22 +332,22 @@ func insertSeries(conn pgxconn.PgxConn, reqs ...copyRequest) (err error) {
 				exemplarLbls = append(exemplarLbls, lvalues)
 			},
 		)
+		if err != nil {
+			visitor.Close()
```

Comment:
why isn't Close simply deferred?

Reply:
We can, but that will lead to insertion into pool after all the `reqs` have completed. But, if we close after each iteration, there is more reusability for the allocated visitors.